devfiles/java-jboss-eap-xp/devfile.yaml

```diff
@@ -0,0 +1,223 @@
+schemaVersion: 2.0.0
+metadata:
+  name: java-jboss-eap-xp
+  version: 1.0.0
+  website: https://access.redhat.com/products/red-hat-jboss-enterprise-application-platform/
+starterProjects:
+  - name: microprofile-config
+    description: EAP XP 2.0 Eclipse Microprofile Config Quickstart
+    git:
+      checkoutFrom:
+        remote: jboss-developer
+        revision: xp-2.0.x
+      remotes:
+        jboss-developer: https://github.com/jboss-developer/jboss-eap-quickstarts.git
+    subDir: microprofile-config
+  - name: microprofile-config
+    description: EAP XP 2.0 Eclipse Microprofile Fault Tolerance Quickstart
+    git:
+      checkoutFrom:
+        remote: jboss-developer
+        revision: xp-2.0.x
+      remotes:
+        jboss-developer: https://github.com/jboss-developer/jboss-eap-quickstarts.git
+    subDir: microprofile-fault-tolerance
+  - name: microprofile-config
+    description: EAP XP 2.0 Eclipse Microprofile Health Quickstart
+    git:
+      checkoutFrom:
+        remote: jboss-developer
+        revision: xp-2.0.x
+      remotes:
+        jboss-developer: https://github.com/jboss-developer/jboss-eap-quickstarts.git
+    subDir: microprofile-health
+  - name: microprofile-config
+    description: EAP XP 2.0 Eclipse Microprofile JWT Quickstart
+    git:
+      checkoutFrom:
+        remote: jboss-developer
+        revision: xp-2.0.x
+      remotes:
+        jboss-developer: https://github.com/jboss-developer/jboss-eap-quickstarts.git
+    subDir: microprofile-jwt
+  - name: microprofile-config
+    description: EAP XP 2.0 Eclipse Microprofile Metrics Quickstart
+    git:
+      checkoutFrom:
+        remote: jboss-developer
+        revision: xp-2.0.x
+      remotes:
+        jboss-developer: https://github.com/jboss-developer/jboss-eap-quickstarts.git
+    subDir: microprofile-metrics
+  - name: microprofile-config
+    description: EAP XP 2.0 Eclipse Microprofile OpenAPI Quickstart
+    git:
+      checkoutFrom:
+        remote: jboss-developer
+        revision: xp-2.0.x
+      remotes:
+        jboss-developer: https://github.com/jboss-developer/jboss-eap-quickstarts.git
+    subDir: microprofile-openapi
+  - name: microprofile-config
+    description: EAP XP 2.0 Eclipse Microprofile OpenTracing Quickstart
+    git:
+      checkoutFrom:
+        remote: jboss-developer
+        revision: xp-2.0.x
+      remotes:
+        jboss-developer: https://github.com/jboss-developer/jboss-eap-quickstarts.git
+    subDir: microprofile-opentracing
+  - name: microprofile-config
+    description: EAP XP 2.0 Eclipse Microprofile REST Client Quickstart
+    git:
+      checkoutFrom:
+        remote: jboss-developer
+        revision: xp-2.0.x
+      remotes:
+        jboss-developer: https://github.com/jboss-developer/jboss-eap-quickstarts.git
+    subDir: microprofile-rest-client
+components:
+  - name: eap-xp
+    container:
+      image: registry.redhat.io/jboss-eap-7/eap-xp2-openjdk11-openshift-rhel8
```

Comment:
Similarly as in previous Devfile, people without Red Hat account won't be able to download this image :-(

Reply:
I guess we can only for now put wildfly examples and not EAP in the public catalog ?

Reply:
but thinking about it - it won't be unusal to have images in private/secured repos - what is the worfklow users would do to use these ? do a `docker login` or similar before running ?

Reply:
@maxandersen I don't think so, the image is only used on the cluster so they would have to grant access to the image registry on their kubernetes

Reply:
> but thinking about it - it won't be unusal to have images in private/secured repos - what is the worfklow users would do to use these ? do a docker login or similar before running ?

Yes, as @ehsavoie pointed out users will have to make sure that the cluster uses the right ImagePullSecret.
For registry.redhat.io on OCP no extra step should be required as they already have this setup. But for Kubernetes, it requires extra steps.

We are trying to target all Kubernetes users, not just OpenShift, that is why we don't want to mix Red Hat only Devfiles, with upstream Devfiles in odo's default public devfile registry. 

We are currently trying to find how to properly handle that so it doesn't create unnecessary complications for users.